contract_sale_payment_mode/models/sale_order.py

```diff
@@ -1,16 +1,43 @@
 # Copyright 2019 ACSONE SA/NV
+# Copyright 2020 ACSONE SA/NV
 # License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
 
-from odoo import api, models
+from odoo import api, fields, models
 
 
 class SaleOrder(models.Model):
     _inherit = 'sale.order'
 
+    contract_payment_mode_id = fields.Many2one(
+        comodel_name='account.payment.mode',
+        string='Contract Payment Mode',
+        domain=[('payment_type', '=', 'inbound')],
+        index=True,
+    )
+    specific_contract_payment_mode = fields.Boolean(
```

Comment:
I don't think adding this field is necessary?

Reply:
it's used for the new field visibility in SO form view